UCSCore/src/main/java/it/cnr/iit/ucs/core/UCSBuilder.java

```diff
@@ -53,42 +42,39 @@
  * This class is in charge of instantiating all the components.
  *
  * @author Antonio La Marra, Alessandro Rosetti
- *
  */
-@Component
-public class UCSRest implements UCSInterface {
-
-    private static final Logger log = Logger.getLogger( UCSRest.class.getName() );
-
-    private AbstractContextHandler contextHandler;
-    private AbstractRequestManager requestManager;
-    private ObligationManagerInterface obligationManager;
-    private SessionManagerInterface sessionManager;
-    private PDPInterface pdp;
-    private PAPInterface pap;
-    private List<PIPBase> pipList = new ArrayList<>();
-    private HashMap<String, PEPInterface> pepMap = new HashMap<>();
+public class UCSBuilder {
 
-    private boolean initialised = false;
+    private static final Logger log = Logger.getLogger( UCSBuilder.class.getName() );
 
-    @Autowired
+    private UCSCore ucsCore;
```

Comment:
sorry to be picky but what is UCSCore?
e.g. if it is a service, then the name should reflect it

Reply:
No, this is actually a good question.
This PR creates the UCSCore project that has no Rest/Spring dependency.

Before it was called "UCSRest" and it also had the build logic mixed with the part to be used by the rest controller.
We have UCSCoreBuilder and UCSCore.
 - UCSCore : Implementation of the UCSInterface and all the components of the UCS.
 - UCSCoreBuilder : it creates an UCSCore.

We can suggest a different name if it is needed. UCSCoreService should be fine but this code has no relation to spring.

Reply:
Yes lets use UCSCoreService. 
And it is fine that there is no link to spring, we can still use the name

Reply:
ok!